prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -761,15 +742,32 @@ static void efa_rdm_ep_destroy_buffer_pools(struct efa_rdm_ep *efa_rdm_ep)
  * Unfinished send includes queued ctrl packets, queued
  * RNR packets and inflight TX packets.
  *
- * @param[in]	efa_rdm_ep	endpoint
- * @return	a boolean
+ * @param[in]  efa_rdm_ep      endpoint
+ * @return     a boolean
  */
 static
 bool efa_rdm_ep_has_unfinished_send(struct efa_rdm_ep *efa_rdm_ep)
 {
-	return !dlist_empty(&efa_rdm_ep->ope_queued_rnr_list) ||
-	       !dlist_empty(&efa_rdm_ep->ope_queued_ctrl_list) ||
-	       (efa_rdm_ep->efa_outstanding_tx_ops > 0);
+	struct dlist_entry *entry, *tmp;
+	struct efa_rdm_ope *ope;
+
+	dlist_foreach_safe(&efa_rdm_ep_domain(efa_rdm_ep)->ope_queued_rnr_list, entry, tmp) {
+		ope = container_of(entry, struct efa_rdm_ope,
+					queued_rnr_entry);
+		if (ope->ep == efa_rdm_ep) {
+			return true;
+		}
+	}
+
+	dlist_foreach_safe(&efa_rdm_ep_domain(efa_rdm_ep)->ope_queued_ctrl_list, entry, tmp) {
+		ope = container_of(entry, struct efa_rdm_ope,
+					queued_ctrl_entry);
+		if (ope->ep == efa_rdm_ep) {
+			return true;
+		}
+	}
+
+    return (efa_rdm_ep->efa_outstanding_tx_ops > 0);
```

Comment:
Would it make sense to check for `efa_rdm_ep->efa_outstanding_tx_ops > 0` before performing the linear search for the EP in the queued lists?

Reply:
Good point. Done